crates/scheduler/src/call/button.rs

```diff
@@ -42,30 +40,35 @@ fn count<B: Board>(call: SchedulerCall<B, api::count::Sig>) {
 
 #[cfg(feature = "board-api-button")]
 fn register<B: Board>(mut call: SchedulerCall<B, api::register::Sig>) {
+    use wasefire_error::{Code, Error};
     let api::register::Params { button, handler_func, handler_data } = call.read();
     let inst = call.inst();
     let result = try {
-        let button = Id::new(*button as usize).map_err(|_| Trap)?;
-        call.scheduler().applet.enable(Handler {
-            key: Key { button }.into(),
-            inst,
-            func: *handler_func,
-            data: *handler_data,
-        })?;
-        board::Button::<B>::enable(button).map_err(|_| Trap)?;
-        Ok(())
+        let button = Id::new(*button as usize)?;
+        call.scheduler()
+            .applet
+            .enable(Handler {
+                key: Key { button }.into(),
+                inst,
+                func: *handler_func,
+                data: *handler_data,
+            })
+            .map_err(|_| Error::internal(Code::InvalidState))?;
```

Comment:
That's actually a user error. And we should actually update `enable` and `disable` to return an `Error` instead of `Trap`. The error here is that the user called `register` twice without calling `unregister` in between.

Reply:
Changed it to an internal error. Do you think `InvalidState` is the correct code here?

Reply:
Yes, that's a good choice.